vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -88,11 +88,11 @@ public void close() {
     requireNonNull(resultHandler, "resultHandler cannot be null");
 
     MongoCollection<JsonObject> coll = getCollection(collection, writeOption);
-    String id = document.getString(ID_FIELD);
+    Object id = document.getValue(ID_FIELD);
     if (id == null) {
       coll.insertOne(document, convertCallback(resultHandler, wr -> document.getString(ID_FIELD)));
```

Comment:
Unfortunately that's going to run afoul of the handler signature.  I tried changing that to Object and then the code generator blew up.  I guess we can insist that id types implement toString and just pass the string representation through to the handler.  In that case I think this should be `wr -> document.getValue(ID_FIELD).toString()`